Sandboxed Play Services Crash

```
type: crash
osVersion: google/oriole/oriole:14/UP1A.231005.007/2023103100:user/release-keys
package: com.google.android.gms:234355038
process: com.google.android.gms.persistent
processUptime: 7202891 + 418 ms
installer: com.android.vending
GmsCompatConfig version: 82

java.lang.RuntimeException: Rethrown by StandaloneSchedulingEngine
	at cmzk.a(:com.google.android.gms@234355038@23.43.55 (190400-577318215):455)
	at cmwk.run(:com.google.android.gms@234355038@23.43.55 (190400-577318215):9)
	at ajlc.c(:com.google.android.gms@234355038@23.43.55 (190400-577318215):50)
	at ajlc.run(:com.google.android.gms@234355038@23.43.55 (190400-577318215):76)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
	at ajqf.run(:com.google.android.gms@234355038@23.43.55 (190400-577318215):8)
	at java.lang.Thread.run(Thread.java:1012)
	Suppressed: dviq: 
		at tk_trace.ScheduledTaskImpl(Unknown Source:0)
		at tk_trace.time-TrustedTimeMaintenanceTaskService_onRunTaskAsync(Unknown Source:0)
Caused by: java.lang.RuntimeException: java.lang.SecurityException: Permission Denial: broadcast from com.google.android.gms asks to run as user -1 but is calling from uid u0a115; this requires android.permission.INTERACT_ACROSS_USERS_FULL or android.permission.INTERACT_ACROSS_USERS
	at bfub.apply(:com.google.android.gms@234355038@23.43.55 (190400-577318215):5)
	at ebua.d(:com.google.android.gms@234355038@23.43.55 (190400-577318215):3)
	at ebub.run(:com.google.android.gms@234355038@23.43.55 (190400-577318215):139)
	... 6 more
Caused by: java.lang.SecurityException: Permission Denial: broadcast from com.google.android.gms asks to run as user -1 but is calling from uid u0a115; this requires android.permission.INTERACT_ACROSS_USERS_FULL or android.permission.INTERACT_ACROSS_USERS
	at android.os.Parcel.createExceptionOrNull(Parcel.java:3066)
	at android.os.Parcel.createException(Parcel.java:3050)
	at android.os.Parcel.readException(Parcel.java:3026)
	at android.os.Parcel.readException(Parcel.java:2968)
	at android.app.IActivityManager$Stub$Proxy.broadcastIntentWithFeature(IActivityManager.java:5778)
	at android.app.ContextImpl.sendBroadcastAsUser(ContextImpl.java:1513)
	at android.app.ContextImpl.sendBroadcastAsUser(ContextImpl.java:1480)
	at android.content.ContextWrapper.sendBroadcastAsUser(ContextWrapper.java:607)
	at android.content.ContextWrapper.sendBroadcastAsUser(ContextWrapper.java:607)
	at avwq.sendBroadcastAsUser(:com.google.android.gms@234355038@23.43.55 (190400-577318215):6)
	at cvrq.a(:com.google.android.gms@234355038@23.43.55 (190400-577318215):39)
	at cvrv.g(:com.google.android.gms@234355038@23.43.55 (190400-577318215):723)
	at cvrr.c(:com.google.android.gms@234355038@23.43.55 (190400-577318215):7)
	at com.google.android.gms.time.singleuserapi.service.TrustedTimeMaintenanceTaskService.hb(:com.google.android.gms@234355038@23.43.55 (190400-577318215):186)
	at bfuc.call(:com.google.android.gms@234355038@23.43.55 (190400-577318215):32)
	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
	... 6 more
Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.server.am.UserController.handleIncomingUser(UserController.java:2457)
	at com.android.server.am.ActivityManagerService.broadcastIntentLockedTraced(ActivityManagerService.java:14521)
	at com.android.server.am.ActivityManagerService.broadcastIntentLocked(ActivityManagerService.java:14431)
	at com.android.server.am.ActivityManagerService.broadcastIntentWithFeature(ActivityManagerService.java:15397)
	at android.app.IActivityManager$Stub.onTransact(IActivityManager.java:2661)


```

Think I had the same issue after updating and rebooting:

```
type: crash
osVersion: google/panther/panther:14/UP1A.231105.003/2023110700:user/release-keys
package: com.google.android.gms:234355038
process: com.google.android.gms.persistent
processUptime: 93509 + 777 ms
installer: com.android.vending
GmsCompatConfig version: 82

java.lang.SecurityException: Permission Denial: broadcast from com.google.android.gms asks to run as user -1 but is calling from uid u0a126; this requires android.permission.INTERACT_ACROSS_USERS_FULL or android.permission.INTERACT_ACROSS_USERS
	at android.os.Parcel.createExceptionOrNull(Parcel.java:3066)
	at android.os.Parcel.createException(Parcel.java:3050)
	at android.os.Parcel.readException(Parcel.java:3026)
	at android.os.Parcel.readException(Parcel.java:2968)
	at android.app.IActivityManager$Stub$Proxy.broadcastIntentWithFeature(IActivityManager.java:5778)
	at android.app.ContextImpl.sendBroadcastAsUser(ContextImpl.java:1513)
	at android.app.ContextImpl.sendBroadcastAsUser(ContextImpl.java:1480)
	at android.content.ContextWrapper.sendBroadcastAsUser(ContextWrapper.java:607)
	at android.content.ContextWrapper.sendBroadcastAsUser(ContextWrapper.java:607)
	at avwq.sendBroadcastAsUser(:com.google.android.gms@234355038@23.43.55 (190400-577318215):6)
	at cvrq.a(:com.google.android.gms@234355038@23.43.55 (190400-577318215):39)
	at cvrs.run(:com.google.android.gms@234355038@23.43.55 (190400-577318215):11)
	at ajlc.c(:com.google.android.gms@234355038@23.43.55 (190400-577318215):50)
	at ajlc.run(:com.google.android.gms@234355038@23.43.55 (190400-577318215):76)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
	at ajqf.run(:com.google.android.gms@234355038@23.43.55 (190400-577318215):8)
	at java.lang.Thread.run(Thread.java:1012)
	Suppressed: dviq: 
		at tk_trace.ScheduledTaskImpl(Unknown Source:0)
		at tk_trace.time-TrustedTimeMaintenanceTaskService_onRunTaskAsync(Unknown Source:0)
Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.server.am.UserController.handleIncomingUser(UserController.java:2457)
	at com.android.server.am.ActivityManagerService.broadcastIntentLockedTraced(ActivityManagerService.java:14540)
	at com.android.server.am.ActivityManagerService.broadcastIntentLocked(ActivityManagerService.java:14450)
	at com.android.server.am.ActivityManagerService.broadcastIntentWithFeature(ActivityManagerService.java:15416)
	at android.app.IActivityManager$Stub.onTransact(IActivityManager.java:2661)


```

--- a/core/java/android/app/ContextImpl.java
+++ b/core/java/android/app/ContextImpl.java
@@ -100,9 +100,9 @@ import android.window.WindowContext;
 import android.window.WindowTokenClient;
 
 import com.android.internal.annotations.GuardedBy;
-import com.android.internal.gmscompat.GmsInfo;
 import com.android.internal.gmscompat.sysservice.GmcPackageManager;
 import com.android.internal.gmscompat.GmsHooks;
+import com.android.internal.gmscompat.sysservice.GmcUserManager;
 import com.android.internal.util.Preconditions;
 
 import dalvik.system.BlockGuard;
@@ -1462,6 +1462,10 @@ class ContextImpl extends Context {
 
     @Override
     public void sendBroadcastAsUser(Intent intent, UserHandle user) {
+        if (GmsCompat.isEnabled()) {
+            user = GmcUserManager.translateUserHandle(user);
+        }
+
         String resolvedType = intent.resolveTypeIfNeeded(getContentResolver());
         try {
             intent.prepareToLeaveProcess(this);
@@ -1485,6 +1489,7 @@ class ContextImpl extends Context {
             Bundle options) {
         if (GmsCompat.isEnabled()) {
             options = GmsHooks.filterBroadcastOptions(intent, options);
+            user = GmcUserManager.translateUserHandle(user);
         }
 
         String resolvedType = intent.resolveTypeIfNeeded(getContentResolver());
@@ -1505,6 +1510,10 @@ class ContextImpl extends Context {
     @Override
     public void sendBroadcastAsUser(Intent intent, UserHandle user,
             String receiverPermission, int appOp) {
+        if (GmsCompat.isEnabled()) {
+            user = GmcUserManager.translateUserHandle(user);
+        }
+
         String resolvedType = intent.resolveTypeIfNeeded(getContentResolver());
         String[] receiverPermissions = receiverPermission == null ? null
                 : new String[] {receiverPermission};
@@ -1542,6 +1551,7 @@ class ContextImpl extends Context {
             Handler scheduler, int initialCode, String initialData, Bundle initialExtras) {
         if (GmsCompat.isEnabled()) {
             options = GmsHooks.filterBroadcastOptions(intent, options);
+            user = GmcUserManager.translateUserHandle(user);
         }
 
         IIntentReceiver rd = null;

--- a/core/java/com/android/internal/gmscompat/sysservice/GmcUserManager.java
+++ b/core/java/com/android/internal/gmscompat/sysservice/GmcUserManager.java
@@ -43,9 +43,18 @@ public class GmcUserManager extends UserManager {
     }
 
     private static void checkUserId(int userId) {
-        if (userId != getUserId()) {
+        if (userId != getUserId() && userId != UserHandle.USER_CURRENT) {
             throw new IllegalStateException("unexpected userId " + userId);
         }
+    }
+
+    public static UserHandle translateUserHandle(UserHandle h) {
+        if (UserHandle.ALL.equals(h)) {
+            return UserHandle.of(getUserId());
+        }
+
+        checkUserId(h.getIdentifier());
+        return h;
     }
 
     private static int getUserSerialNumber() {

